### PR TITLE
Support views in BigQuery source

### PR DESCRIPTION
**Pull request by ardian4**
Added: example pipeline when table is view

JIRA:
https://issues.cask.co/browse/PLUGIN-171

### CODE REVIEW

File: src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySourceConfig.java [hunk 1]
```diff
@@ -106,9 +111,31 @@
...
+  @Macro
+  @Nullable
+  @Description("Whether to allow querying views. Since BigQuery views are not materialized by default,"
+    + " querying them may have a performance overhead.")
+  private String enableQueryingViews;
+
+  @Name(NAME_VIEW_MATERIALIZATION_PROJECT)
+  @Macro
+  @Nullable
+  @Description("The project name where the View should be materialized. "
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/gcp/bigquery/source/BigQuerySourceConfig.java]: nit: View -> view

**Review by rmstar — APPROVED** [review]

File: src/main/java/io/cdap/plugin/gcp/bigquery/source/PartitionedBigQueryInputFormat.java [hunk 2]
```diff
@@ -142,6 +155,48 @@ private String generateQuery(String partitionFromDate, String partitionToDate, S
...
+   */
+  private TableReference createExportTableReference(
+    Type type, String inputProjectId,
+    String datasetId,
+    String tableId,
+    Configuration configuration) {
+
+    TableReference tableReference = new TableReference().setTableId(tableId);
+
+    if (type == Type.VIEW) {
```
Comment (rmstar) [inline src/main/java/io/cdap/plugin/gcp/bigquery/source/PartitionedBigQueryInputFormat.java]: Don't we need to check for `type == Type.MATERIALIZED_VIEW` here?

**Review by rmstar — COMMENTED** [review]
please squash commits once all changes are done.

**Comment by ardian4** [comment]
> please squash commits once all changes are done.

@rmstar all changes are done and squashed.

**Pull request merged**